src/BloomExe/Edit/EditingModel.cs

```diff
@@ -509,6 +515,29 @@ void OnPageSelectionChanged(object sender, EventArgs e)
 			GC.Collect();//i put this in while looking for memory leaks, feel free to remove it.
 		}
 
+		private void CheckForUnsavedOrigamiAndSave()
+		{
+			// The following does the equivalent of these 3 lines of javascript
+			// marginBox.removeClass('origami-layout-mode');
+			// marginBox.find('.bloom-translationGroup .textBox-identifier').remove();
+			// fireCSharpEditEvent('preparePageForEditingAfterOrigamiChangesEvent', '');
```

Comment:
1. I hate to have 15 lines of C# to do what would be three lines of JavaScript. If C# is the only place we can detect this is needed, couldn't we still arrange to call into JavaScript? Failing that,...
2. It's not obvious that RethinkPageAndReloadIt() is equivalent to firing preparePageForEditingAfterOrigamiChangesEvent. Maybe that method is exactly what the event is raised? But what if we later attach some additional behavior to the event? Could we more closely simulate raising it?